### ISSUE TITLE
AutoUpload not working, battery optimization bug

### ISSUE THREAD

**Issue by samedamci** [nextcloud/android#4815]
### Actual behaviour
- When I want to enable auto upload feature is displayed info to disable battery optimization for NextCloud app, but this function is disabled, and sync not working.

### Expected behaviour
- Auto upload should works.

### Environment data
Android version:
Android 9
Device model: 
Xiaomi Mi A2 Lite
Stock or customized system:
Lineage OS 16
Nextcloud app version:
3.9.0 & dev 20191102
Nextcloud server version:
16.0.3
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

**Comment by eliezerb2** [comment]
I have the same issue

**Comment by mhzawadi** [comment]
I get the same issue on a Oneplus 3T running Android 9

**Comment by garygitton** [comment]
Same issue with samsung s9 +

**Comment by eyriewow** [comment]
Same issue on my Xiaomi Mi A1 on Android Version 9

**Comment by nextgen-networks** [comment]
My device is affected too (Xiaomi Mi 9 SE and Android 10/Q)

**Comment by tihomircolev** [comment]
Same issue Xiaomi redmi 5

**Comment by miovee** [comment]
same. Battery optimization warning. not syncing 
Device : Galaxy Fold

**Comment by setaou** [comment]
Same issue here with Xiaomi Mi 9T (== Redmi K20)

**Comment by BrandonMcInroy** [comment]
Same issue with S10, everything up to date.  Battery optimization is off for app.

**Comment by TsingJyujing** [comment]
Same issue on my HUAWEI P9+

**Comment by ScottHammett** [comment]
Same problem.  Wants me to exclude nextcloud from battery optimization to auto upload.  There is no way to do that on Android 5.0.1

**Comment by chdorb** [comment]
Same issue on HUAWEI P20, Android 9, EMUI 9.1.0 (latest)
Android Nexcloud 3.9.2

**Comment by msundin** [comment]
Auto upload stopped working  completely after Galaxy S10 was update from Android 9 to Android 10.

**Comment by szaimen** [comment]
have you looked here?
https://dontkillmyapp.com/

**Comment by adojck** [comment]
Same issue with Samsung Galaxy S10+
Android 10
Nextcloud Android app 3.9.2

Every time I try to set-up sync for a folder I see "Battery Optimization" window with "Close" and "Disable" buttons. I tried pushing each, nothing changes, sync does not work.

**Comment by markuskramerIgitt** [comment]
No AutoUpload with 
- Nextcloud client 3.9.2
- Lineage 16.0 2020-01-05
- No power saver, app battery not optimized
- Sony XA2

**Comment by mjflower** [comment]
Same issue here. Mi A1 android 9 everything up to date. Auto sync does not start. Battery optimisation is off for the app.

**Comment by sp4ke** [comment]
Same here, battery optimization disabled:

Huawei P20 Pro
Nextcloud Dev Version 20200125
EMUI 9.0.0
Android 9

**Comment by ivanrojasorg** [comment]
Same issue.

Xioami Mi A1 - Android 9 

Nextcloud Client Android - 3.10.1

Nextcloud Server - 17.0.2

**Comment by ivanrojasorg** [comment]
Then of 2 days searching the way to have a funtionallity AutoUpload, I finded one way.

1. First, test configure a folder like camara. Then you take a picture. If it create a folder new and upload the image, you app is ok. The app only will upload the new files. You will have upload manually the old files.

2. If step 1 dont working. Download the last app Nextcloud Dev. https://f-droid.org/en/packages/com.nextcloud.android.beta/ Install the app and test again de step 1.

Please tell me if this steps works for you.

**Comment by exili0n** [comment]
Hi, noticed same problem today. In nextcloud dev the alert still presists, but syncing is OK. 

> Then of 2 days searching the way to have a funtionallity AutoUpload, I finded one way.
> 
>     1. First, test configure a folder like camara. Then you take a picture. If it create a folder new and upload the image, you app is ok. The app only will upload the new files. You will have upload manually the old files.
> 
>     2. If step 1 dont working. Download the last app Nextcloud Dev. https://f-droid.org/en/packages/com.nextcloud.android.beta/ Install the app and test again de step 1.
> 
> 
> Please tell me if this steps works for you.

**Comment by mjflower** [comment]
tried the dev app last week. same issue.

**Comment by exili0n** [comment]
Hi, on nextcloud dev the same issue (

**Comment by tobiasKaminsky** [comment]
Thanks for replying, @exili0n -> closing

**Comment by MorgothSauron** [comment]
AutoUpload doesn't work for me
- Android version: Android 9
- Device model: Xiaomi Redmi Note 8 Pro
- Stock or customized system: Stock
- Nextcloud app version: 3.11.0
- Nextcloud server version: 18.0.3

No matter how long I way autoupload never upload new pictures. In parallel Dropbox works fine. Uploading the same files manually work.

**Comment by QuakyCZ** [comment]
I also have the same issue on these devices:

- Samsung Galaxy A20e
- Samsung Galaxy A50

**Comment by johackim** [comment]
Yes ! I confirm, the dev version work : https://f-droid.org/en/packages/com.nextcloud.android.beta/ :smiley:

**Comment by alphaw1n** [comment]
On my Mi8 I have the same issue. No actions on phone or in the system can't fix it. I am very disappointed, that such a critical issue is still presented. Seems I need to search for more stable solutions.

- Mi8
- Android 10
- MIUI 12.0.1 Global Stable
- Nextcloud 3.18.0

**Comment by alphaw1n** [comment]
> Thanks for replying, @exili0n -> closing

Please, reopen this issue, the problem is still presented.

**Comment by LeandroSQ** [comment]
Same here on a S20 Fe
Android 12
OneUI 4.1

**Comment by CiozZ** [comment]
Same issue, on a Samsung Galaxy S9

**Comment by kyuuk** [comment]
Same problem for Samsung Galaxy s9 
Application version : 3.20.3

**Comment by jcarrut2** [comment]
Same problem for Samsung Galaxy S9+
Application version: 3.20.3, fresh install.
This bug is not resolved. Please re-open.

**Comment by githubtmp7** [comment]
Same problem for Samsung Galaxy S9
Application version: 3.21.1.
This bug is not resolved.

**Comment by flicker581** [comment]
Same problem on Redmi 5 Plus, MIUI 11.0.2, Android 8.1.0. Tried to give every found permission, tried to install dev versions. First time when I enable autosync, the app asks for additional permission to work at background. After that, the same error about battery optimization is displayed but no permission dialogue after it. Autosync does not work at all.

But on another phone, Poco M4, which runs Android 11, the same message about battery optimization is also displayed each time. But autosync is working fine. So, I suppose these are two different bugs.

**Comment by reos-rcrozier** [comment]
Same problen Samsung Galaxy A21s, Android 12

Excluded app fro battery opt, excluded from RAM opt. No syncing.

**Comment by mdio33** [comment]
Just decided to try NextCloud this week.
First test on a iPhone, no problem at all.

But with two differents Androids i've got this error.

In a Samsung A50 stock rom, Android 11, One UI 31, i've got that error with Play or F-Droid version and does not upload anything!

In an old LeEco Le2 with custom rom LineageOS, i've got the error, but the upload starts anyway, even found new folders (Skype) and asked if wanted to upload the pictures.

It been 3 years since this issue was reported... does anyone from Nexcloud see this??

**Comment by daburger** [comment]
Samsung Galaxy S21 Ultra - same issue.
Unable to auto-sync as it keeps asking me to disable battery optimization for this app, even though it already is.

**Comment by Taeleus** [comment]
Same problem here. It worked for a moment, then I stopped it and changed the folder location to a better one on the NAS, now it wont sync anything. No matter what setting I check/uncheck.

**Comment by Taeleus** [comment]
soooo. It randomly works now. *shrug*

**Comment by kalashnikovmk** [comment]
same issue with xiaomi mi8 app_version: 3.24.2 server_version: 26.0.1_1.6.27 (run in TrueNAS )

**Comment by Kylar514** [comment]
Same issue with the S21+.  Can't auto upload, I get the battery optimization message.

**Comment by kftX** [comment]
Samsung Galaxy S23 Ultra here (Android 13)

Auto-upload works but can't change any settings. If I try I'll get a warning about battery optimisation and it won't change any settings. I can't change battery optimisation since this OneUI version has no such menu.

Nextcloud already has unrestricted access to my battery in app settings, it should not give this error, making the custom upload literally unusable.

![Screenshot_20230718_200740_Settings](https://github.com/nextcloud/android/assets/47950410/339449ab-c73b-4f80-a05e-4ad1e967eed1)

![Screenshot_20230718_200945_Nextcloud](https://github.com/nextcloud/android/assets/47950410/43362d87-fa34-4f8a-99aa-3c00cabadf63)

**Comment by Astrrra** [comment]
Having the same issue on Galaxy Z Fold 4, Android 13, same symptoms as in the comment above.

**Comment by ctrenkada** [comment]
Same problem on my Samsung Galaxy A54. How come this hasn't been solved in 4 years?

**Comment by zheka2** [comment]
Same problem on Samsung Galaxy S21 FE

**Comment by Taeleus** [comment]
> Same problem here. It worked for a moment, then I stopped it and changed the folder location to a better one on the NAS, now it wont sync anything. No matter what setting I check/uncheck and no matter what destination folder I set. Samsung Galaxy Note 9.

I commented in a decent timeframe but its been working just fine for a long time now. It still warns me about the battery settings even though I have that warning turned off and have the battery settings correct, but things do upload and go where they say theyre going to go. Its been nice actually just mindlessly backing up all my photos. Very happy with it.

I think it just had to take a day or two and re-run some background tasks and sync up its settings again. I did turn off sync, deleted all the files/folders on the server, setup a new folder and then pointed all the auto-upload settings at the new folder. Set those settings and walked away from it. 2 days later its uploaded everything and is now auto uploading pictures as I take them (at night when the phone is plugged in and charging is when I have it set to sync)

**Comment by daburger** [comment]
Both mine and my wife's phone have the same issue. Battery optimization turned off but still warns me....but sync remains off and resets configuration.

**Comment by Tobiaswec** [comment]
**I had the same issue but u found the solution (Samsung S10e) !**

First go into Apps --> Nextcloud --> deactivate "Remove permission if app is unused"
![photo_2023-09-08_18-19-31](https://github.com/nextcloud/android/assets/22745978/222734e8-4505-4555-96b3-227afbf24dc6)

Then in the same manner: Apps --> Nextcloud --> Battery --> set to "Unrestricted"
![photo_2023-09-08_18-19-38](https://github.com/nextcloud/android/assets/22745978/30c9f28a-f0e6-4a9d-bb7f-6d9c08954770)

I hope this helps! :)

**Comment by muzhaqi16** [comment]
Same problem on my Samsung s22 Ultra, Battery is set to Unrestricted and remove permissions if app is unused is disabled.

**Comment by chheller** [comment]
I was able to work around this on certain folders. I followed all of the above for my S22 Ultra on Android 13, One UI 5.1 against a fresh NextCloud plugin installed via TrueNAS on my local network.
- Setting Battery to Unrestricted
- Disabling "Remove Permissions if unused" 
- Granting all permissions
- Disabling Battery Saver Protection in NextCloud
- Disabling Battery Saver on my phone
- Configuring the Performance mode to Maximum Performance

Then, I navigated to the Auto-upload settings and enabled folders. However, when prompted by the modal dialog to disable battery optimizations, rather than pressing `close` or `disable`, I would press _outside_ of the modal. This seems to enable auto upload. However, I still have not had any success with my Camera folder which has 6.4k photos. Trying to upload it manually results in an app crash after some time. Trying to configure it for auto upload might cause the auto upload to stop working. I also had a lot of conflicts, for no clear reason. When I dismissed the notifications for conflicts, auto upload seemed to stop. 

In the event that auto upload stopped running, I could go to app's settings page > storage > and then cleared the cache. Subsequently restarting the app seemed to kick off the auto upload again.

**Comment by muzhaqi16** [comment]
Tried this again on my Samsung S22 ultra and Samsung S20 5G and this issue is still present, I removed the app, deleted cache and data, made sure that it is set to unrestricted on the power and still the same issue. I noticed that when I set to upload only on unmetered network even though i unchecked it later it was still waiting for it.

**Comment by alex005005** [comment]
Same here on S22 Ultra. No way to get it working. This is a crucial feature to consider nextcloud a real replacement for Google Services.

**Comment by Taeleus** [comment]
> Same issue here on S22 Ultra. For whatever reason it started syncing after I brought up the App settings via Settings -> Apps -> Nextcloud and not via Long-Tap -> App-Info. Didn't even change anything in permissions or battery settings. Just opened the App Settings and immediately started syncing.

Check my earlier comment. I forget what I did but mine works fine now. Think I literally deleted all the folders and made a new folder and set it up again from scratch. It still warns me about the battery but its been syncing fine for 4-6 months now without issues.

**Comment by shermy** [comment]
Let's please jump into fixing this issue Nextcloud, same here on Samsung Galaxy A10s

**Comment by YamiOdymel** [comment]
same issue with S24

**Comment by ties-pibc** [comment]
Had the same issue.

Resolved this by adding going to device management and changing the processing speed, this was set to optimized. 
When I changed this to maximum it started working again.

Not sure on the actual naming in English since my phone is in Dutch.
But screen is called "Apparaatonderhoud" and then the option "Verwerkingssnelheid" under the "Prestaties" (performance)  tab.

**Comment by YamiOdymel** [comment]
@ties-pibc I guess you are talking about Performance Profile? Sadly change to "Max Performance" it didn't work for me

![image](https://github.com/user-attachments/assets/a59fa143-39bd-47db-ad47-29722c16de9a)

![image](https://github.com/user-attachments/assets/c897a67a-f325-49a4-9c74-949edc2becf9)

**Comment by ties-pibc** [comment]
Hi there,

No unfortunately it is not that. I changed my UI to English. This was the setting 

![Screenshot_20240910_091519_Device care](https://github.com/user-attachments/assets/4f4d82da-d84f-4ee5-aadc-092e5aeee105)


Under device care it is the processing speed option to maximum what did the trick for me.

**Comment by oyvigri** [comment]
Ive tried all the settings and combinations as this thread suggests to no avail. I think I might have go for alternative sync solution.
Im so happy that nextcloud exists with a team of passionate people behind it, and I truly do love all of them, but maaaan every single step of the way from installing, configuring and daily usage is just such a huuuge pain with this software. 
Nextcloud needs constant intervention which is exhausting

**Comment by souvannavongalex** [comment]
Reviving this thread into 2025. Had the same issue syncing as most explained in this post. Noticed that my auto upload to Nextcloud stopped sometime last month. My device is a Samsung S23 Ultra.
Ensured that my device has battery optimization turned off and nextcloud was unhindered by my device's settings, no luck with auto upload.

Though, I will mention I went for the nuclear option of just uninstalled nexcloud completely from my phone and reinstalled it. Reconfigured the auto uploaded folders, and ensured batter optimization was turned off and syncing has resumed from where it left off.

TLDR: I uninstall the Nextcloud App & Reinstalled it. Your mileage may vary

**Comment by TaZi-loh** [comment]
> Reviving this thread into 2025. Had the same issue syncing as most explained in this post. Noticed that my auto upload to Nextcloud stopped sometime last month. My device is a Samsung S23 Ultra. Ensured that my device has battery optimization turned off and nextcloud was unhindered by my device's settings, no luck with auto upload.

I have exactly this same problem with my S24.

**Comment by ties-pibc** [comment]
Small update on this.
I used to have a virusscanner on my phone (Bitdefender), recently i switched to a different scanner and since then i have 0 problems.

**Comment by tntturtl** [comment]
Nextcloud autoupload broke down on Samssung Galaxy A53 after recent phone's firmware update.

**Comment by Dark-Delu** [comment]
Same problem with S21 FE. I tried all suggested steps already. 
No battery restrictions and my phones max power mode enabled.

**Comment by ra1u** [comment]
I also have this issue. I am running LinageOS and have noted that i have no syncing since jan 2025 when i have upgraded my LineageOS. Is there an option to sync and upload files manually?

**Comment by ngdangtu-vn** [comment]
Unfortunately, my phone got the same problem as well even though samsung promise to follow the standard...
